File: backend/finance-service/src/main/java/com/bytecode/finance/model/FeeRecord.java
package com.bytecode.finance.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Document(collection = "fee_records")
public class FeeRecord {
    @Id
    private String id;
    private String studentEmail;
    private Double totalAmount;
    private Double paidAmount;
    private Double balanceAmount;
    private String status; // PAID, PARTIAL, PENDING
    private Date lastPaymentDate;
}

File: lms-node-backend/src/models/FeeRecord.js
const mongoose = require('mongoose');

const FeeRecordSchema = new mongoose.Schema({
    studentEmail: { type: String, required: true },
    totalAmount: { type: Number, required: true },
    paidAmount: { type: Number, default: 0 },
    balanceAmount: { type: Number, required: true },
    status: { type: String, enum: ['PAID', 'PARTIAL', 'PENDING'], default: 'PENDING' },
    lastPaymentDate: { type: Date, default: Date.now },
    createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('FeeRecord', FeeRecordSchema);
